### ISSUE TITLE
MA0077 `Equals(TSelf)` without `IEquatable<TSelf>` being raised on `ref struct`s when targeting < .NET 9

### ISSUE THREAD

**Issue by YoshiRulz** [meziantou/Meziantou.Analyzer#823]
### Version of the Meziantou.Analyzer NuGet package

2.0.210

### Rule Identifier

MA0077

### Target Framework

netstandard2.0, net48

### C# Language version

C# 13

### Description

While `ref struct`s can now implement interfaces, the type parameter would be the struct itself, which isn't allowed.

### Reproduction Steps

Minimal code:

```c#
public readonly ref struct TestStruct(int n) //: IEquatable<TestStruct>
{
	public readonly int N
		=> n;

	public readonly override bool Equals(object? other)
		=> false;

	public readonly bool Equals(TestStruct other)
		=> n == other.N;

	public readonly override int GetHashCode()
		=> n;
}
```

**Comment by meziantou** [comment]
I'm not sure to understand the issue. Your repro code compiles without any error, even when adding the interface.

**Comment by YoshiRulz** [comment]
[Repro project](https://github.com/YoshiRulz/meziantou-analyzer-823-repro)

**Comment by meziantou** [comment]
This is a limitation of C# 12. If you update to C# 13, the project builds without error.

That's something the analyzer can check

**Comment by YoshiRulz** [comment]
Oh, that's my bad, I forgot to `nix-shell -p dotnetCorePackages.dotnet_10.sdk` so the project was created with the .NET 8 SDK.
Fixing that, the repro project does indeed compile, but it relies on the [added `allows ref struct` constraint](https://github.com/dotnet/runtime/commit/eb14da93fe4fd29cc80eba6b12c6452ea10d166e) in the .NET 9 BCL. I changed the project to target .NET Framework, like my real project, and now it's failing.
